docs/src/main/asciidoc/kafka-streams-guide.adoc

```diff
@@ -0,0 +1,1141 @@
+
+[source]
+----
+org.acme.quarkus.sample.kafkastreams.bootstrap.servers=localhost:9092
+----
+
+== Building and Running the Applications
+
+We now can build the `producer` and `aggregator` applications:
+
+[source, subs=attributes+]
+----
+mvn clean package -f producer/pom.xml
+mvn clean package -f aggregator/pom.xml
+----
+
+Instead of running them directly on the host machine using the Quarkus dev mode,
+we're going to package them into container images and launch them via Docker Compose.
+This is done in order to demonstrate scaling the `aggregator` aggregation to multiple nodes later on.
+
+The `Dockerfile` created by Quarkus by default needs one adjustment for the `aggregator` application in order to run the Kafka Streams pipeline.
+To do so, edit the file `aggregator/src/main/docker/Dockerfile.jvm` and replace the line `FROM fabric8/java-alpine-openjdk8-jre` with `FROM fabric8/java-centos-openjdk8-jdk`.
+
+Next create a Docker Compose file (`docker-compose.yaml`) for spinning up the two applications as well as Apache Kafka and ZooKeeper like so:
+
+[source, yaml]
+----
+version: '3.5'
+
+services:
+  zookeeper:
+    image: strimzi/kafka:0.11.3-kafka-2.1.0
+    command: [
+      "sh", "-c",
+      "bin/zookeeper-server-start.sh config/zookeeper.properties"
+    ]
+    ports:
+      - "2181:2181"
+    environment:
+      LOG_DIR: /tmp/logs
+    networks:
+      - kafkastreams-network
+  kafka:
+    image: strimzi/kafka:0.11.3-kafka-2.1.0
+    command: [
+      "sh", "-c",
+      "bin/kafka-server-start.sh config/server.properties --override listeners=$${KAFKA_LISTENERS} --override advertised.listeners=$${KAFKA_ADVERTISED_LISTENERS} --override zookeeper.connect=$${KAFKA_ZOOKEEPER_CONNECT} --override num.partitions=$${KAFKA_NUM_PARTITIONS}"
+    ]
+    depends_on:
+      - zookeeper
+    ports:
+      - "9092:9092"
+    environment:
+      LOG_DIR: "/tmp/logs"
+      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://kafka:9092
+      KAFKA_LISTENERS: PLAINTEXT://0.0.0.0:9092
+      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
+      KAFKA_NUM_PARTITIONS: 3
+    networks:
+      - kafkastreams-network
+
+  producer:
+    image: quarkus-quickstarts/kafka-streams-producer:1.0
+    build:
+      context: producer
+      dockerfile: src/main/docker/Dockerfile.${QUARKUS_MODE:-jvm}
+    environment:
+      KAFKA_BOOTSTRAP_SERVERS: kafka:9092
+    networks:
+      - kafkastreams-network
+
+  aggregator:
+    image: quarkus-quickstarts/kafka-streams-aggregator:1.0
+    build:
+      context: aggregator
+      dockerfile: src/main/docker/Dockerfile.${QUARKUS_MODE:-jvm}
+    environment:
+      ORG_ACME_QUARKUS_SAMPLE_KAFKASTREAMS_BOOTSTRAP_SERVERS: kafka:9092
+    networks:
+      - kafkastreams-network
+
+networks:
+  kafkastreams-network:
+    name: ks
+----
+
+To launch all the containers, building the `producer` and `aggregator` container images,
+run `docker-compose up --build`.
```

Comment:
Should the command only be `docker-compose up` which will build the image when necessary as opposed to `docker-compose up --build` which will build the image each time the command is launched ?

Reply:
I think it's better for these instructions to rather build once too often than forgetting it once. E.g. if you re-do them a second time, you might easily use the old images unintentionally.